cmd/gateway-handlers.go

```diff
@@ -509,9 +512,13 @@ func (api gatewayAPIHandlers) DeleteBucketPolicyHandler(w http.ResponseWriter, r
 		return
 	}
 
-	// Delete bucket access policy, by passing an empty policy
-	// struct.
+	// Delete bucket access policy.
 	objAPI.DeleteBucketPolicies(bucket)
+
+	globalBucketPolicies.SetBucketPolicy(bucket, policyChange{
+		true, policy.BucketAccessPolicy{},
+	})
```

Comment:
I would suggest you to remove the bucket policy related changes we decided to do it differently and not keep/manage any in memory state for gateway code.  So you can perhaps just remove my PR in this regard.  We can say currently bucket policies do not work on Sia backend and fix it later.

Reply:
So you want me to remove all the changes that you made to support bucket policies? There are multiple servers running now by Sia users that depend on bucket policy support. I guess they could continue using a branch of my version of Minio until policies are fully supported, but I do imagine you'll get quite a bit of feedback from users regarding policies not working. :)

Reply:
Yes the problem is not that, the problem is that its not a correct approach and has lots of holes in it.  It is okay to loose this ability while we address the overall architecture.  I reached out to you on slack, perhaps we can discuss on other things in the PR.